### ISSUE TITLE
`Unordered::next` does not return a `FusedFuture`

### ISSUE THREAD

**Issue by daprilik** [udoprog/unicycle#16]
`futures::stream::StreamExt::next` returns a future that [explicitly implements `FusedFuture`](https://docs.rs/futures-util/0.3.21/src/futures_util/stream/stream/next.rs.html#22-26). 
`Unordered::next` ought to as well.

This comes up when using `.next()` with the `futures::select!` macro.

**Comment by udoprog** [comment]
I'd have no issue to include it behind a feature flag (e.g. `futures03`).

In the meantime, consider using the [`tokio::select!` macro](https://docs.rs/tokio/latest/tokio/macro.select.html) which doesn't require [FusedFuture].

**Comment by udoprog** [comment]
Actually just saw that we already implement `Stream` behind the `futures-rs` feature flag which is set by default. So that would be the one to use. But it also means we already provide some degree of futures 0.3 compat so you can use `StreamExt::next(&mut futures)` to poll it in a `futures::select!` macro.

**Comment by daprilik** [comment]
Yeah, after poking around a bit more, I also realized that I could use `StreamExt::next`.  While this works, it's pretty wordy, and looks really out of place in more complex chained expressions. It would be nice if the inherent impl also implemented `FusedFuture` so folks could use `.next()` directly.

So yep, I agree - putting this behind the `futures-rs` feature flag seems like the right play.

Seems like the impl would essentially be a straight mirror of the [impl in futures-rs](https://docs.rs/futures-util/0.3.21/src/futures_util/stream/stream/next.rs.html#22-26). Is this something you could merge in yourself, or should I send in a PR?

**Comment by udoprog** [comment]
Please do a PR!
